### PR TITLE
Create the "Using the Infinispan Client" guide

### LINKED ISSUE #30457: Split the actual Infinispan Client documentation Guide
### Description

The Guide is too long. Split the actual Infinispan documentation in 2 separate guides
- Infinispan Client Guide
- infinispan Extension Reference Guide 

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by karesti**
* `infinispan-client-reference.adoc` has the content of the old infinispan-client.adoc (no deep content review needed)
* `infinispan-client.adoc` is the new **Using the Infinispan Client** guide file that needs review

Closes https://github.com/quarkusio/quarkus/issues/30457
Quickstart PR:  https://github.com/quarkusio/quarkus-quickstarts/pull/1254

### CODE REVIEW

File: docs/src/main/asciidoc/infinispan-client-reference.adoc [hunk 1]
```diff
@@ -0,0 +1,751 @@
...
+.build.gradle
+----
+implementation 'io.quarkus:quarkus-infinispan-client'
+annotationProcessor 'org.infinispan.protostream:protostream-processor:4.6.1.Final' <1>
+----
+<1> Mandatory in the Gradle build to enable the generation of the files in the annotation based serialization
+
+== Configuring the Infinispan client
+
+Open the `application.properties` file in the `src/main/resources` directory with any text editor.
```
[the remark this answers is not in the document]
Comment (karesti) [reply docs/src/main/asciidoc/infinispan-client-reference.adoc:49]: yes, I forgot to remove this part

**Comment by karesti** [comment]
@wburns I updated the ref guide with the changes. would you mind checking the content of the infinispan client guide please? the one is related to the changes of the quickstart as well
thanks 🙏🏻

File: docs/src/main/asciidoc/images/dev-ui-infinispan.png
Comment (gastaldi) [inline docs/src/main/asciidoc/images/dev-ui-infinispan.png:1]: It would be nice if the guide used the new Dev UI card

Comment (karesti) [reply docs/src/main/asciidoc/images/dev-ui-infinispan.png:1]: I added that one in the Infinispan client guide, I can add it to the ref guide too

Comment (gastaldi) [reply docs/src/main/asciidoc/images/dev-ui-infinispan.png:1]: Makes sense

**Review by wburns — APPROVED** [review]

**Pull request merged**